lib/active_resource/base.rb

```diff
@@ -1283,14 +1330,27 @@ def persisted?
       @persisted
     end
 
+    # TODO remove: indifferent_access causes duplicates values and loss of reference
+    def attributes
+      super.with_indifferent_access
+    end
```

Comment:
There are [existing tests asserting that the `Hash` returned by `#attributes` supports indifferent access](https://github.com/rails/activeresource/blob/9c8a2ee30325ff0bf789a239cc25fa361ec4a887/test/cases/base/load_test.rb#L136-L141). The value returned by `ActiveModel::Attributes#attributes` only supports String keys.

Transforming the value returned from `super` with a call to `Hash#with_indifferent_access` preserves the behavior covered by the tests, but _deeply_ transforms the `Hash`, which has unintended side effects for attributes that are `Hash` instances or `Hash`-like.

Is there a way to limit the indifferent access to be top-level only, leaving nested values unaffected?